src/sentry/api/endpoints/project_transaction_threshold.py

```diff
@@ -0,0 +1,102 @@
+import math
+
+from django.db import transaction
+from rest_framework import serializers, status
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases.project import ProjectEndpoint, ProjectSettingPermission
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models import transaction_threshold
+from sentry.models.transaction_threshold import TRANSACTION_METRICS, ProjectTransactionThreshold
+
+
+class ProjectTransactionThresholdSerializer(serializers.Serializer):
+    threshold = serializers.IntegerField()
+    metric = serializers.CharField()
+
+    def validate_metric(self, metric):
+        for key, value in TRANSACTION_METRICS.items():
+            if value == metric:
+                return key
+
+        raise serializers.ValidationError(f"Invalid transaction metric - {metric}")
+
+    def validate_threshold(self, threshold):
+        return int(math.ceil(threshold / 100)) * 100
```

Comment:
Should prevent users from adding granular in the UI, but this is just an extra precaution

Reply:
Would it make more sense to 400 in this situation rather than silently converting the threshold they set? Then validation on the front end should try to avoid this situation where possible.